tnc/data_handler.py

```diff
@@ -1830,20 +1914,25 @@ def burst_watchdog(self):
         DATA BURST
         """
         # IRS SIDE
+        # TODO: We need to redesign this part for cleaner state handling
         if not static.ARQ_STATE or static.ARQ_SESSION_STATE != 'connected' or static.TNC_STATE != 'BUSY' or not self.is_IRS:
-            return
-
+            # return only if not ARQ STATE and not ARQ SESSION STATE as they are different use cases
+            if not static.ARQ_STATE and static.ARQ_SESSION_STATE == 'disconnected' or not self.is_IRS:
+                return
+        # we want to reach this state only if connected ( == return above not called )
```

Comment:
This seems to fix #189 
However, I'm not 100% happy with this. We might check the "arq session" feature in a separate test and optimizing this area then as well.